origin_ui/origin_test.go

```diff
@@ -0,0 +1,354 @@
+		assert.JSONEq(t, `{"msg":"Success"}`, recorderLoginWithNewPassword.Body.String())
+	})
+
+	//Invoking password reset without a cookie should result in failure
+	t.Run("Without valid cookie", func(t *testing.T) {
+		resetPayload := `{"password": "newpassword"}`
+		reqReset, err := http.NewRequest("POST", "/api/v1.0/origin-ui/resetLogin", strings.NewReader(resetPayload))
+		assert.NoError(t, err)
+
+		reqReset.Header.Set("Content-Type", "application/json")
+
+		recorderReset := httptest.NewRecorder()
+		router.ServeHTTP(recorderReset, reqReset)
+
+		//Check ok http reponse
+		assert.Equal(t, 403, recorderReset.Code)
+		//Check that success message returned
+		assert.JSONEq(t, `{"error":"Password reset only available to logged-in users"}`, recorderReset.Body.String())
+	})
+
+}
+
+func TestPasswordBasedLoginAPI(t *testing.T) {
+	///////////////////////////SETUP///////////////////////////////////
+	//Add an admin user to file to configure
+	content := "admin:password\n"
+	_, err := tempPasswdFile.WriteString(content)
+	assert.NoError(t, err, "Error writing to temp password file")
+
+	//Configure UI
+	err = ConfigureOriginUI(router)
+	assert.NoError(t, err)
+
+	//Create a user for testing
+	err = writePasswordEntry("user", "password")
+	assert.NoError(t, err, "error writing a user")
+	password := "password"
+	user := "user"
+	///////////////////////////////////////////////////////////////////
+
+	//Invoke with valid password, should get a cookie back
+	t.Run("Successful Login", func(t *testing.T) {
+		payload := fmt.Sprintf(`{"user": "%s", "password": "%s"}`, user, password)
+
+		//Create a request
+		req, err := http.NewRequest("POST", "/api/v1.0/origin-ui/login", strings.NewReader(payload))
+		assert.NoError(t, err)
+
+		req.Header.Set("Content-Type", "application/json")
+
+		recorder := httptest.NewRecorder()
+		router.ServeHTTP(recorder, req)
+		//Check ok http reponse
+		assert.Equal(t, http.StatusOK, recorder.Code)
+		//Check that success message returned
+		assert.JSONEq(t, `{"msg":"Success"}`, recorder.Body.String())
+		//Check for a cookie being returned
+		cookies := recorder.Result().Cookies()
+		foundCookie := false
+		for _, cookie := range cookies {
+			if cookie.Name == "login" {
+				foundCookie = true
+			}
+		}
+		assert.True(t, foundCookie)
+	})
+
+	//Invoke without a password should fail
+	t.Run("Without password", func(t *testing.T) {
+		payload := fmt.Sprintf(`{"user": "%s"}`, user)
+		//Create a request
+		req, err := http.NewRequest("POST", "/api/v1.0/origin-ui/login", strings.NewReader(payload))
+		assert.NoError(t, err)
+		req.Header.Set("Content-Type", "application/json")
+
+		recorder := httptest.NewRecorder()
+		router.ServeHTTP(recorder, req)
+		//Check http reponse code 401
+		assert.Equal(t, 401, recorder.Code)
+		assert.JSONEq(t, `{"error":"Login failed"}`, recorder.Body.String())
+	})
+
+	//Invoke with invalid user, should fail
+	t.Run("Without user", func(t *testing.T) {
```

Comment:
You have tests for "Without user" and "Without password" but what about "Incorrect user" (where the user doesn't exist) or "Wrong password"

Reply:
Added with new commit